extensions/control-plane/provision/provision-bigquery/src/test/java/org/eclipse/edc/connector/provision/gcp/BigQueryProvisionerTest.java

```diff
@@ -171,23 +171,17 @@ private void provisionSucceeds(String serviceAccountName)  throws IOException {
             useAdc = true;
             // When using defuault credentials, createBigQuery is executed passing a null argument.
             when(bqFactory.createBigQuery(null)).thenReturn(bigQuery);
-
-            serviceAccount = BigQueryProvisioner.ADC_SERVICE_ACCOUNT;
-            serviceAccountName = serviceAccount.getName();
-
-            when(iamService.createDefaultAccessToken()).thenReturn(token);
+            serviceAccount = IamService.ADC_SERVICE_ACCOUNT;
         } else {
```

Comment:
as we discussed in the previous PR, this big test with conditions in it is not maintainable, I took the freedom to split it in 2 tests to explain what I had in mind

```java
    private final BigQueryFactory bqFactory = mock();
    private final BigQueryProvisioner bigQueryProvisioner = new BigQueryProvisioner(gcpConfiguration, bqFactory, iamService, monitor);

    @Test
    void provisionSuccessUsingAdc() throws IOException {
        var serviceAccount = IamService.ADC_SERVICE_ACCOUNT;
        var token = getTestToken();
        when(bqFactory.createBigQuery(any())).thenReturn(bigQuery);
        when(iamService.getServiceAccount(null)).thenReturn(serviceAccount);
        when(iamService.createAccessToken(any())).thenReturn(token);
        var table = mock(Table.class);
        when(table.exists()).thenReturn(true);
        when(bigQuery.getTable(TEST_TARGET.getTableId())).thenReturn(table);
        var resourceDefinition = BigQueryResourceDefinition.Builder.newInstance()
                .id(RESOURCE_ID)
                .transferProcessId(TRANSFER_ID)
                .property(BigQueryServiceSchema.PROJECT, TEST_PROJECT)
                .property(BigQueryServiceSchema.DATASET, TEST_DATASET)
                .property(BigQueryServiceSchema.TABLE, TEST_TABLE)
                .property(BigQueryServiceSchema.CUSTOMER_NAME, CUSTOMER_NAME)
                .build();

        var result = bigQueryProvisioner.provision(resourceDefinition, Policy.Builder.newInstance().build());

        assertThat(result).succeedsWithin(1, SECONDS)
                .extracting(StatusResult::getContent).satisfies(response -> {
                    assertThat(response.getResource()).isInstanceOfSatisfying(BigQueryProvisionedResource.class, resource -> {
                        assertThat(resource.getId()).isEqualTo(RESOURCE_ID);
                        assertThat(resource.getResourceDefinitionId()).isEqualTo(RESOURCE_ID);
                        assertThat(resource.getTransferProcessId()).isEqualTo(TRANSFER_ID);
                        assertThat(resource.getResourceName()).isEqualTo(TEST_TABLE + "-table");
                        assertThat(resource.getProject()).isEqualTo(TEST_PROJECT);
                        assertThat(resource.getDataset()).isEqualTo(TEST_DATASET);
                        assertThat(resource.getTable()).isEqualTo(TEST_TABLE);
                        assertThat(resource.getServiceAccountName()).isEqualTo(IamService.ADC_SERVICE_ACCOUNT.getName());
                        assertThat(resource.hasToken()).isTrue();
                    });

                    assertThat(response.getSecretToken()).usingRecursiveComparison().isEqualTo(token);
                });
        verify(iamService).createAccessToken(serviceAccount);
        verify(bqFactory).createBigQuery(null);
    }

    @Test
    void provisionSuccessUsingServiceAccount() throws IOException {
        var serviceAccount = new GcpServiceAccount(TEST_EMAIL, TEST_SERVICE_ACCOUNT_NAME, TEST_DESCRIPTION);
        var token = getTestToken();
        when(bqFactory.createBigQuery(TEST_SERVICE_ACCOUNT_NAME)).thenReturn(bigQuery);
        when(iamService.getServiceAccount(TEST_SERVICE_ACCOUNT_NAME)).thenReturn(serviceAccount);
        when(iamService.createAccessToken(serviceAccount)).thenReturn(token);
        var table = mock(Table.class);
        when(table.exists()).thenReturn(true);
        when(bigQuery.getTable(TEST_TARGET.getTableId())).thenReturn(table);
        var resourceDefinition = BigQueryResourceDefinition.Builder.newInstance()
                .id(RESOURCE_ID)
                .transferProcessId(TRANSFER_ID)
                .property(BigQueryServiceSchema.PROJECT, TEST_PROJECT)
                .property(BigQueryServiceSchema.DATASET, TEST_DATASET)
                .property(BigQueryServiceSchema.TABLE, TEST_TABLE)
                .property(BigQueryServiceSchema.CUSTOMER_NAME, CUSTOMER_NAME)
                .property(BigQueryServiceSchema.SERVICE_ACCOUNT_NAME, TEST_SERVICE_ACCOUNT_NAME)
                .build();

        var result = bigQueryProvisioner.provision(resourceDefinition, Policy.Builder.newInstance().build());

        assertThat(result).succeedsWithin(1, SECONDS)
                .extracting(StatusResult::getContent).satisfies(response -> {
                    assertThat(response.getResource()).isInstanceOfSatisfying(BigQueryProvisionedResource.class, resource -> {
                        assertThat(resource.getId()).isEqualTo(RESOURCE_ID);
                        assertThat(resource.getResourceDefinitionId()).isEqualTo(RESOURCE_ID);
                        assertThat(resource.getTransferProcessId()).isEqualTo(TRANSFER_ID);
                        assertThat(resource.getResourceName()).isEqualTo(TEST_TABLE + "-table");
                        assertThat(resource.getProject()).isEqualTo(TEST_PROJECT);
                        assertThat(resource.getDataset()).isEqualTo(TEST_DATASET);
                        assertThat(resource.getTable()).isEqualTo(TEST_TABLE);
                        assertThat(resource.getServiceAccountName()).isEqualTo(TEST_SERVICE_ACCOUNT_NAME);
                        assertThat(resource.hasToken()).isTrue();
                    });

                    assertThat(response.getSecretToken()).usingRecursiveComparison().isEqualTo(token);
                });
        verify(iamService).createAccessToken(serviceAccount);
        verify(bqFactory).createBigQuery(TEST_SERVICE_ACCOUNT_NAME);
    }
```

Reply:
I like how 'satisfies' makes the following comparison simpler, but I am not sure about the code duplication, even though it is a test. This is about two ifs versus the code in the test that must be then updated in both test cases.

Reply:
I have currently simplified the test, the two ifs (3+1 LOCs) are quite manageable in my opinion, please let me know if this works for you, thanks!

Reply:
sorry, but having conditions into a test is an anti-pattern, they must be simple and explain the logic they are gonna test in a straightforward way. 
they should not contain any additional logic because that will make them prone to bugs, and a bug in a test is hard to detect and it could lead to sneaky bugs in production code.

Please remove the conditionals in the tests

Reply:
Sounds good, I will update (and eventually use your code :)

Reply:
Conditions removed, kept the nice use of 'satisfies' as well, thank you!